src/components/discover-sheet/ListsSection.js

```diff
@@ -0,0 +1,334 @@
+            genericAssets[toLower(item.address)],
+            nativeCurrency
+          )
+      );
+    } else {
+      if (!lists?.length) return [];
+      const currentList = lists.find(list => list.id === selectedList);
+      if (!currentList) {
+        return [];
+      }
+
+      return currentList.tokens.map(
+        address =>
+          ethereumUtils.getAsset(allAssets, toLower(address)) ||
+          ethereumUtils.formatGenericAsset(
+            genericAssets[toLower(address)],
+            nativeCurrency
+          )
+      );
+    }
+  }, [
+    allAssets,
+    favorites,
+    genericAssets,
+    lists,
+    nativeCurrency,
+    selectedList,
+  ]);
+
+  const handlePress = useCallback(
+    item => {
+      navigate(
+        ios ? Routes.EXPANDED_ASSET_SHEET : Routes.EXPANDED_ASSET_SCREEN,
+        {
+          asset: item,
+          fromDiscover: true,
+          longFormHeight: initialChartExpandedStateSheetHeight,
+          type: 'token',
+        }
+      );
+    },
+    [navigate]
+  );
+
+  const itemProps = useMemo(
+    () => ({
+      showAddButton: true,
+      showBalance: false,
+    }),
+    []
+  );
+
+  const renderItem = useCallback(
+    ({ item: list, index }) => (
+      <ListButton
+        key={`list-${list.id}`}
+        onPress={() => handleSwitchList(list.id, index)}
+        selected={selectedList === list.id}
+      >
+        <Row>
+          <Emoji name={list.emoji} size="small" />
+          <ListName
+            color={
+              selectedList === list.id
+                ? colors.alpha(colors.blueGreyDark, 0.8)
+                : colors.alpha(colors.blueGreyDark, 0.5)
+            }
+            lineHeight="paragraphSmall"
+            size="lmedium"
+            weight="bold"
+          >
+            {list.name}
+          </ListName>
+        </Row>
+      </ListButton>
+    ),
+    [colors, handleSwitchList, selectedList]
+  );
+
+  return (
+    <Column>
+      <Flex paddingHorizontal={19}>
+        <Text size="larger" weight="heavy">
+          Lists
+        </Text>
+      </Flex>
+
+      <Fragment>
+        <Column>
+          <FlatList
+            contentContainerStyle={{
+              paddingBottom: 6,
+              paddingHorizontal: 19,
+              paddingTop: 10,
+            }}
+            data={listData}
+            getItemLayout={getItemLayout}
+            horizontal
+            keyExtractor={item => item.id}
+            ref={listRef}
+            renderItem={renderItem}
+            scrollsToTop={false}
+            showsHorizontalScrollIndicator={false}
+          />
+          <EdgeFade />
+        </Column>
+        <Column>
+          {!ready ? (
+            times(2, index => (
+              <AssetListItemSkeleton
+                animated
+                descendingOpacity
+                key={`skeleton-pools-${index}`}
+              />
+            ))
+          ) : listItems?.length ? (
+            listItems
+              .filter(item => {
```

Comment:
Just for prettiness: would be nice to do this listItems filter outside of the component logic